### PR TITLE
Move contextFromString out of JaegerSpanContext into TextMapCodec

### LINKED ISSUE #518: Move internal method contextFromString out of JaegerSpanContext
## Requirement - what kind of business use case are you trying to solve?

Currently, the internal `contextFromString` method resides in the `JaegerSpanContext` class. It produces a `JaegerSpanContext` instance from a string that conforms with the `TextMapCodec`. Arguably, this is not an ideal location for it to reside.

## Problem - what in Jaeger blocks you from solving the requirement?

In the future, we would like to support other codecs, such as W3C trace identifiers (https://github.com/jaegertracing/jaeger/issues/855). This method implies the only valid codec is the `TextMapCodec` seeing as it is the canonical method for using strings to produce `JaegerSpanContext` and implicitly relies on the `TextMapCodec` format.

## Proposal - what do you suggest to solve the problem or improve the existing situation?

Move the `contextFromString` method into the `TextMapCodec` class instead.

### PR DESCRIPTION
**Pull request by isaachier**
Fixes #518.

### CODE REVIEW

**Comment by isaachier** [comment]
As usual, Travis fails due to crossdock timeout.

File: jaeger-core/src/main/java/io/jaegertracing/internal/propagation/TextMapCodec.java [hunk 1]
```diff
@@ -54,6 +57,25 @@ private TextMapCodec(Builder builder) {
     this.baggagePrefix = builder.baggagePrefix;
   }
 
+  public static JaegerSpanContext contextFromString(String value)
```
Comment (jpkrohling) [inline jaeger-core/src/main/java/io/jaegertracing/internal/propagation/TextMapCodec.java]: If it's used only in tests, can't we make this private and test the inject operation instead?

Comment (yurishkuro) [reply jaeger-core/src/main/java/io/jaegertracing/internal/propagation/TextMapCodec.java]: This has been blocking the other PR, so I would rather not cause major refactoring of the tests. But going for minimum visibility is always a good goal. In this case, wouldn't all the tests work just as well if this was package level viz?

Comment (isaachier) [reply jaeger-core/src/main/java/io/jaegertracing/internal/propagation/TextMapCodec.java]: I tried it and sadly no. `propagation` is its own package.

Comment (yurishkuro) [reply jaeger-core/src/main/java/io/jaegertracing/internal/propagation/TextMapCodec.java]: Ah, yes. Ok then.

Comment (isaachier) [reply jaeger-core/src/main/java/io/jaegertracing/internal/propagation/TextMapCodec.java]: I can probably fix the other tests. Don't merge yet.

Comment (yurishkuro) [reply jaeger-core/src/main/java/io/jaegertracing/internal/propagation/TextMapCodec.java]: Oops too late. Put another one if a simple fix.

**Pull request merged**